OpenRA.Mods.Common/Traits/RevealsMap.cs

```diff
@@ -0,0 +1,76 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Reveals shroud and fog across the whole map while active.")]
+	public class RevealsMapInfo : ConditionalTraitInfo
+	{
+		[Desc("Stance the watching player needs to see the shroud removed.")]
+		public readonly Stance ValidStances = Stance.Ally;
```

Comment:
It may be a problem with my implementation of this trait but the Stances are ignored. In a team MP match only the player that activates RevealsMap can see the map when ValidStances: Ally is used.

Reply:
The logic here looks completely bogus. What is this field supposed to do, reveal the map for other players that have this stance?

Reply:
Yes, that is my take on it; I robbed some parts of RevealMapCrateAction to try & fix it. https://github.com/Inq8/CAmod/commit/aafe335bdd09cae9473ecb7dbcad28bd9c33ed66

Reply:
I filed #18610 which seems to work for me.